File: LeetCode/Simulation/JS/Leetcode2283-check-if-number-has-equal-digit-count-and-digit-value.js
var digitCount = function(num) {
    let mp = new Array(10).fill(0);
    for (const x of num) {
        mp[x.charCodeAt() - '0'.charCodeAt()]++;
    }
    for (let i = 0;i < num.length;i++) {
        const cnt = parseInt(num[i]);
        if (mp[i] != cnt) {
            return false;
        }
    }
    return true;
};

File: LeetCode/Simulation/Golang/Leetcode2283-check-if-number-has-equal-digit-count-and-digit-value.go
package sbgo

func digitCount(num string) bool {
  mp := make([]int, 10)
  for _, c := range num {
    mp[c - '0']++
  }
  for i := 0;i < len(num);i++ {
    cnt := num[i] - '0'
    if int(cnt) != mp[i] {
      return false
    }
  }
  return true
}
